src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -58,35 +62,53 @@
      * The tag of the colonies.
      */
     private static final String                     TAG_COLONIES          = "colonies";
+
     /**
      * The tag of the pseudo unique identifier
      */
     private static final String                     TAG_UUID              = "uuid";
 
     /**
+     * The Tag to store the recipes to NBT.
+     */
+    private static final String TAG_RECIPES = "recipes";
+
+    /**
      * The damage source used to kill citizens.
      */
     private static final DamageSource               CONSOLE_DAMAGE_SOURCE = new DamageSource("Console");
     /**
      * The list of all colonies.
      */
     @NotNull
-    private static final ColonyList<Colony>         colonies              = new ColonyList<>();
+    private static final ColonyList<Colony>         colonies        = new ColonyList<>();
+
     /**
      * The list of all colonies by world.
      */
     @NotNull
-    private static final Map<Integer, List<Colony>> coloniesByWorld       = new HashMap<>();
+    private static final Map<Integer, List<Colony>> coloniesByWorld = new HashMap<>();
+
     /**
      * The list of colony views.
      */
     @NotNull
     private static final ColonyList<ColonyView>     colonyViews           = new ColonyList<>();
 
     /**
+     * Map of all recipes which have been discovered globally already.
+     */
+    private static final Map<IToken, RecipeStorage> recipes = new HashMap<>();
```

Comment:
IToken being from the Request System, correct?

Reply:
Exactly

Reply:
Cool